third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -195,13 +195,17 @@ impl LazyLoadedFunction {
 
     /// If the function hasn't been resolved (loaded) yet, loads it. The gas is also charged for
     /// function loading and any other module accesses.
-    pub(crate) fn as_resolved<RTTCheck: RuntimeTypeCheck>(
+    pub(crate) fn as_resolved<LoaderImpl, RTTCheck>(
         &self,
-        module_storage: &impl ModuleStorage,
-        layout_converter: &LayoutConverter<impl Loader>,
+        loader: &LoaderImpl,
```

Comment:
Hmm. Wondering whether this shows that the dispatch macro could be avoided?

Reply:
What do you mean by "this"? We can propagate it up (what I do in this PR), but there is an entrypoint at some point where you need to dispatch?

Reply:
this can be simply impl Loader right?

Reply:
updated to `impl Loader`